### PR TITLE
Make heartbeat timeouts and intervals configurable

### PR DESCRIPTION
**Pull request by DanielCosme**
This PR will make configurable the activity heartbeats and throttle intervals in Temporal workflows.

### CODE REVIEW

File: enduro.toml [hunk 1]
```diff
@@ -83,3 +83,9 @@ disabled = true
 
 [metadata]
 processNameMetadata = false
+
+[worker]
+heartbeatThrottleInterval = "5m"
+
+[activity]
+heartbeatTimeout = "1h"
```
Comment (jraddaoui) [inline enduro.toml:91]: We should use the actual defaults when possible and change them in their env config.

File: internal/workflow/policies.go [hunk 2]
```diff
@@ -36,6 +36,7 @@ func withActivityOptsForLongLivedRequest(ctx temporalsdk_workflow.Context) tempo
 // The activity is responsible for returning a NRE error.
 func withActivityOptsForHeartbeatedRequest(ctx temporalsdk_workflow.Context, heartbeatTimeout time.Duration) temporalsdk_workflow.Context {
 	return temporalsdk_workflow.WithActivityOptions(ctx, temporalsdk_workflow.ActivityOptions{
+		ScheduleToStartTimeout: forever,
```
Comment (jraddaoui) [inline internal/workflow/policies.go:39]: It defaults to unlimited already.

File: main.go [hunk 3]
```diff
@@ -255,13 +255,15 @@ func main() {
...
+			MaxHeartbeatThrottleInterval:      config.Worker.HeartbeatThrottleInterval,
+			DefaultHeartbeatThrottleInterval:  config.Worker.HeartbeatThrottleInterval,
 		})
 		if err != nil {
 			logger.Error(err, "Error creating Temporal worker.")
 			os.Exit(1)
 		}
 
-		w.RegisterWorkflowWithOptions(workflow.NewProcessingWorkflow(h, colsvc, pipelineRegistry, logger).Execute, temporalsdk_workflow.RegisterOptions{Name: collection.ProcessingWorkflowName})
+		w.RegisterWorkflowWithOptions(workflow.NewProcessingWorkflow(h, colsvc, pipelineRegistry, logger, workflow.Config{ActivityHeartbeatTimeout: config.Activity.HeartbeatTimeout}).Execute, temporalsdk_workflow.RegisterOptions{Name: collection.ProcessingWorkflowName})
```
Comment (jraddaoui) [inline main.go:266]: I'd reuse the workflow config.

File: main.go [hunk 4]
```diff
@@ -391,12 +393,22 @@ type configuration struct {
...
 	Hooks map[string]map[string]interface{}
 }
 
+type WorkerConfig struct {
+	HeartbeatThrottleInterval time.Duration
+}
+
+type ActivityConfig struct {
+	HeartbeatTimeout time.Duration
+}
```
Comment (jraddaoui) [inline main.go:410]: Instead of adding a new one.

File: enduro.toml (hunk 1 above)
Comment (jraddaoui) [inline enduro.toml:91]: After that change, this would be:

```
[workflow]
activityHeartbeatTimeout = "1h"
```

Like we did in other projects.

**Review by jraddaoui — CHANGES_REQUESTED** [review]
Some minor comments @DanielCosme.

File: enduro.toml (hunk 1 above)
Comment (DanielCosme) [reply enduro.toml]: Done. f936fa7df08e26e9a20df1403f315a2ff8b32fce

File: internal/workflow/policies.go (hunk 2 above)
Comment (DanielCosme) [reply internal/workflow/policies.go]: Deleted.

File: main.go (hunk 3 above)
Comment (DanielCosme) [reply main.go]: Done.

File: main.go (hunk 4 above)
Comment (DanielCosme) [reply main.go]: Yes.

File: enduro.toml (hunk 1 above)
Comment (DanielCosme) [reply enduro.toml]: Done.

**Review by jraddaoui — APPROVED** [review]
Thanks @DanielCosme!

**Pull request merged**